src/assignments/assignments.service.ts

```diff
@@ -153,4 +165,21 @@ export class AssignmentsService {
     });
     return null;
   }
+
+  async buildAssignmentResponse(
+    assignments: AssignmentDocument[],
```

Comment:
Can we overload the function so it takes both the array and the single version?

Reply:
We can, but I think it's easier to always receive and respond with an array and none of the rest of the code needs to change